tests/scalers_go/external_scaler_sj/external_scaler_sj_test.go

```diff
@@ -166,3 +158,84 @@ func testScaleDown(t *testing.T, kc *kubernetes.Clientset, data templateData) {
 	assert.True(t, WaitForJobCount(t, kc, testNamespace, 0, 60, 1),
 		"job count should be 0 after 1 minute")
 }
+
+func TestMinReplicaCount(t *testing.T) {
+	kc := GetKubernetesClient(t)
+	minReplicaCount := 2
+	maxReplicaCount := 50
+	metricValue := 0
+
+	data, templates := getTemplateData(minReplicaCount, maxReplicaCount, metricValue)
+
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	assert.True(t, WaitForJobCount(t, kc, testNamespace, 2, 60, 1),
+		"job count should be 2 after 1 minute")
+
+	DeleteKubernetesResources(t, kc, testNamespace, data, templates)
+}
+
+func TestMinReplicaCountGreaterMaxReplicaCountScalesOnlyToMaxReplicaCount(t *testing.T) {
```

Comment:
We prefer to have one test case per e2e test. Apart from that, could you add this tests as a part of `internals`? Just copy this could be enough (renaming the namespaces) but we prefer to keep scalers test for testing scalers and internals/secrets-provider tests to test those things, just for having them split and categorized